jena-fuseki2/jena-fuseki-webapp/src/test/java/org/apache/jena/fuseki/TestAdmin.java

```diff
@@ -278,7 +278,29 @@
         deleteDataset(dsTest);
     }
 
-    // Sync task testing
+    @Test public void sleep_1() {
+        String x = execSleepTask(null, 1);
+    }
+
+    @Test public void sleep_2() {
+        try {
+            String x = execSleepTask(null, -1);
+            fail("Sleep call unexpectedly succeed");
+        } catch (HttpException ex) {
+            assertEquals(400, ex.getStatusCode());
+        }
+    }
+
+    @Test public void sleep_3() {
+        try {
+            String x = execSleepTask(null, 20*1000+1);
+            fail("Sleep call unexpectedly succeed");
+        } catch (HttpException ex) {
+            assertEquals(400, ex.getStatusCode());
+        }
+    }
+
+    // Async task testing
```

Comment:
Do we want some additional test cases that validate that we can now have multiple simultaneous async tasks?  I see test cases that cover JENA-1871 but not any that cover JENA-1872

Reply:
Tricky. Any ideas how to?

I can investigate reading the tasks list for active tasks but we have situations where loaded CI servers do weird things (pauses of 10s of seconds or longer - I guess the VM isn't get real CPU time) so "overlap " is hard to ensure in a reliable way, including with the client code in the `@Test` itself.

Having "works mostly" is bad -  it it then natural to ignore test failures because they become "expected".

I have tested locally, including hitting the limit.

Reply:
I've pushed two tests.

If they become unreliable in practice, I'd prefer that we comment out the `@Test`.

Reply:
LGTM